gtda/utils/validation.py

```diff
@@ -278,7 +280,7 @@ def check_point_clouds(X, distance_matrices=False, **kwargs):
         for i, x in enumerate(X):
             try:
                 xnew = _check_array_mod(x, **kwargs_, ensure_2d=True)
-                if distance_matrices:
+                if distance_matrices and not issparse(xnew):
```

Comment:
Reason for this: Sparse matrices can easily end up being non-square, but that is OK in all current applications for which `distance_matrices` might be set to `True`, i.e. the persistent homology transformers in `gtda.homology.simplicial``.